unison-cli/src/Unison/Share/Sync.hs

```diff
@@ -624,10 +631,13 @@ completeTempEntities httpClient unisonShareUrl connect repoInfo downloadedCallba
           pure (Left (SyncError err))
         Right (Share.DownloadEntitiesSuccess entities) -> do
           downloadedCallback (NESet.size hashes)
-          atomically do
-            writeTQueue entitiesQueue (hashes, entities)
-            recordNotWorking workerCount
-          pure (Right ())
+          case validateEntities entities of
+            Left err -> pure . Left . SyncError . Share.DownloadEntitiesEntityValidationFailure $ err
```

Comment:
Just out of curiosity, what does this error look like?

Reply:
Good question!  Here's me cloning base with an empty map of known mismatches:



```
.> clone @unison/base

  ❗️

  The hash associated with the given term component entity is incorrect.

  The associated hash is:
  #4h9sgktb6jb2u9ubblo5pj3jl9nkjvr2coa8u0fkt0hjl8v1i26jppbu190fmrt8tbts3dfmqm4ohdeeq52k5hv93rt6auvaacsniig
  The computed hash is:
  #243evnehcbtmk52049o5938vijh89pi75cvu82nev5esi42c4vokvjj7fgj29tc77nmpupktmomht34to2a4i498mcdju3f753gi6mg

  Please create an issue and report this to the Unison team.
```